Assets/Scripts/Combat/Damageable.cs

```diff
@@ -283,71 +323,89 @@ private void ColorChangeForLaser()
         }
     }
 
-    private void FireDamage(bool isResistant, bool isWeak)
+    private void FireDamage(bool isResistant, bool isWeak, int index, int damage)
     {
-        if (DoesShowDamageParticles()) { _fireParticles.Play(); }
-
-        int fireDamage = 0;
-
-        if ((!isResistant && !isWeak) || (isResistant && isWeak)) { fireDamage = 8; }
-
-        if (isResistant && !isWeak) { fireDamage = 4; }
-
-        if (isWeak && !isResistant) { fireDamage = 12; }
-
+        int fireDamage = CalculateDamageForTypes(isResistant, isWeak, index, damage);
+        
         if (fireDamage == 0) { return; }
 
+        if (DoesShowDamageParticles()) { _fireParticles.Play(); }
+
         if (_fireRoutine != null) { StopCoroutine(_fireRoutine); }
 
-        _fireRoutine = StartCoroutine(Afterburn(fireDamage));
+        _timer = 0;
+        _fireRoutine = StartCoroutine(Afterburn(fireDamage, index));
     }
 
-    private void ElectricDamage(bool isDamageChain)
+    private int ElectricDamage(bool isDamageChain, bool isResistant, bool isWeak, int index, int damage)
     {
-        if (_isBeingElectrocuted && isDamageChain) { return; }
+        int electricDamage = CalculateDamageForTypes(isResistant, isWeak, index, damage);
+
+        if (_isBeingElectrocuted && isDamageChain) { return 0; }
 
         _isBeingElectrocuted = true;
 
-        if (TryGetComponent<BaseStateMachine>(out BaseStateMachine baseStateMachine)) { baseStateMachine.CanMove = false; }
+        if (TryGetComponent(out BaseStateMachine baseStateMachine)) { baseStateMachine.CanMove = false; }
```

Comment:
Instead of doing `TryGetComponent(out BaseStateMachine baseStateMachine)` fire an `OnElectrocuted` event that `BaseStateMachine` listens for, and then fires a function sets `CanMove` to false. Always try to remove any 3rd party class dependencies

Reply:
For the record, I didn't wrote that.